src/pal/src/exception/signal.cpp

```diff
@@ -554,6 +555,23 @@ static void common_signal_handler(PEXCEPTION_POINTERS pointers, int code,
         ASSERT("sigprocmask failed; error is %d (%s)\n",errno, strerror(errno));
     } 
 
+#if defined(_AMD64_)
+    if (pointers->ExceptionRecord->ExceptionCode == EXCEPTION_INT_DIVIDE_BY_ZERO)
+    {
+        // It is possible that an overflow may also have been mapped to a divide-by-zero
+        // exception. This happens when we try to divide the maximum negative value of a
+        // signed integer with -1. 
+        //
+        // Thus, we will attempt to decode the instruction @ RIP to determine if that
+        // is the case using the faulting context.
+        if (IsDivByZeroAnIntegerOverflow(pointers->ContextRecord, pointers->ExceptionRecord))
```

Comment:
Debugger does not care about the distinction between divide by zero and overflow.

The dangerous part I had in mind is that somebody else's code will use instruction encodings that this code does not handle properly.